File: fidesz.py
from random import randint, choice

propaganda = [ #kifogások listája
    "A brüsszeli szankciók tönkretesznek minket!",
    "Brüsszel az ön adójából fizetné a háborút!",
    "Ne táncoljunk úgy, ahogy ők fütyülnek!",
    "Ők adót emelnek, és ukrán aranybudira költik a pénzed.",
    "Magyarország előre megy, nem hátra!",
    "Dühíti önt Brüsszel?"
]

def fidesz(a,n): #fidesz sort
    maxIdx = 0 
    for i in range(n): #leggazdagabb (oligarcha) megkeresése
        if a[i]>a[maxIdx]:
            maxIdx = i
    for i in range(n): #mindenki más meglopása
        if i!=maxIdx:
            a[maxIdx]+= a[i]
            a[i] = 0
    print(choice(propaganda)) #az ellenzék hibáztatása

n = 16
a = []
for i in range(n): 
    a.append(randint(0,99))
print(*a)
fidesz(a,n)
print(*a)

# Tényleges szortírozás nem valósul meg, pont ahogy
# a fidesz alatt sem történik megfelelő kormányzás,
# viszont cserébe mindenki meg lesz lopva.

File: fidesz.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

const char* propaganda[] = {
    "A brüsszeli szankciók tönkretesznek minket!",
    "Brüsszel az ön adójából fizetné a háborút!",
    "Ne táncoljunk úgy, ahogy ők fütyülnek!",
    "Ők adót emelnek, és ukrán aranybudira költik a pénzed.",
    "Magyarország előre megy, nem hátra!",
    "Dühíti önt Brüsszel?"
};
int propagandaSzam = 6;

void fidesz(int* a, int n);

int main()
{
    srand(time(NULL));
    
    int n = 16;
    int* a = malloc(sizeof(int)*n);
    for(int i=0; i<n; i++) a[i] = rand() % 100;
    if(a==NULL) return 1;

    for(int i=0; i<n; i++)
    {
        printf("%d ",a[i]);
    }
    printf("\n");
    
    fidesz(a,n);
    
    for(int i=0; i<n; i++)
    {
        printf("%d ",a[i]);
    }
    printf("\n");

    free(a);
}

void fidesz(int* a, int n)
{
    int maxIdx = 0;
    
    for(int i=0; i<n; i++)
    {
        if(a[i]>a[maxIdx])
        {
            maxIdx = i;
        }
    }
    for(int i=0; i<n; i++)
    {
        if(i!=maxIdx)
        {
            a[maxIdx]+= a[i];
            a[i] = 0;
        }
    }
    printf("%s\n",propaganda[rand() % propagandaSzam]);


}